<a href="https://colab.research.google.com/github/Emiliewu/MachineLearning-Assignments/blob/main/week5_d1_demo_ml_Train_Test_Split.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Train Test Split (Model Validation)

## import

In [5]:
import pandas as pd
fname = "/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week05/Data/cali_housing.csv"
df = pd.read_csv(fname)
df.head()

MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  Latitude  \
0  8.3252      41.0  6.984127   1.023810       322.0  2.555556     37.88   
1  8.3014      21.0  6.238137   0.971880      2401.0  2.109842     37.86   
2  7.2574      52.0  8.288136   1.073446       496.0  2.802260     37.85   
3  5.6431      52.0  5.817352   1.073059       558.0  2.547945     37.85   
4  3.8462      52.0  6.281853   1.081081       565.0  2.181467     37.85   

   Longitude  MedHouseVal  
0    -122.23        4.526  
1    -122.22        3.585  
2    -122.24        3.521  
3    -122.25        3.413  
4    -122.25        3.422

* The entire dataframe contains 20640 entries You can confirm this with the code and output here:

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   MedInc       20640 non-null  float64
 1   HouseAge     20640 non-null  float64
 2   AveRooms     20640 non-null  float64
 3   AveBedrms    20640 non-null  float64
 4   Population   20640 non-null  float64
 5   AveOccup     20640 non-null  float64
 6   Latitude     20640 non-null  float64
 7   Longitude    20640 non-null  float64
 8   MedHouseVal  20640 non-null  float64
dtypes: float64(9)
memory usage: 1.4 MB


* For this data set, MedHouseVal is the target we are trying to predict. For now, let's just keep all of the rest of the columns as potential features.

In [3]:
# The target we are trying to predict
y = df['MedHouseVal']
# The features we will use to make the prediction
X = df.drop(columns = 'MedHouseVal')

* Once you have loaded in your data frame and split it into the target (y) and features (X), you can split it into a training set and a test set.

In [4]:
# Import the TTS from sciki-learn
from sklearn.model_selection import train_test_split
# Train test split
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

* We are defining 4 new variables here: 

* From our original X dataframe, we get two new dataframes:

    - X_train is the data we will use to train the model and contains only the features
    - X_test is the data we set aside for testing (evaluating) our model and contains only features
* From our original y series, we get two new series:

    - y_train contains the target values for the data corresponding to the X_train features
    - y_test contains the target values for the data we have set aside for testing corresponding to the X_test feautres
Note that a random state was set to 42. Since the split is a random process (some rows of data end up in the training set and some rows of data end up in the test set), we can ensure that we can get reproducible results by assigning a random state. Any number can be used here.

To verify the split, you can check the length of each:

In [6]:
len(X_train)

15480

* The output above shows that of the original 20640 rows, there are 15480 rows in the training set. (This is 75% of the values)

In [7]:
len(X_test)

5160

* The output above shows that of the original 20640 rows, there are 5160 rows in the test set. (This is 25% of the values)

You should have the same values for your target y. The code below verifies this.

In [8]:
len(y_train)

15480

In [9]:
len(y_test)

5160